ui/selectors/selectors.js

```diff
@@ -396,6 +396,20 @@ export function getAddressBookEntryOrAccountName(state, address) {
   return entry && entry.name !== '' ? entry.name : address;
 }
 
+export function getAccountName(identities, address) {
+  const entry = Object.values(identities).find((identity) =>
+    isEqualCaseInsensitive(identity.address, toChecksumHexAddress(address)),
+  );
+  return entry && entry.name !== '' ? entry.name : '';
+}
+
+export function getMetadataContractName(address) {
```

Comment:
We should use `getTokenList` here, so that we use the correct list if `isTokenDetectionInactiveOnMainnet` is true

Reply:
@danjm Done [here](https://github.com/MetaMask/metamask-extension/pull/16018/commits/3efda348801dc6c2df0eca58c405556895c51e7f).